frontend/ui/widget/notification.lua

```diff
@@ -28,6 +32,15 @@ local Notification = InputContainer:new{
     toast = true, -- closed on any event, and let the event propagate to next top widget
 
     _nums_shown = {}, -- array of stacked notifications
+
+    SOURCE_NONE = 0,
+    SOURCE_BOTTOM_MENU_ICON = 1,
+    SOURCE_BOTTOM_MENU_TOGGLE = 2,
+    SOURCE_BOTTOM_MENU_FINE = 4,
+    SOURCE_BOTTOM_MENU_MORE = 8,
+    SOURCE_DISPATCHER = 16,
+    SOURCE_GESTURE = 32,
```

Comment:
You're not yet using SOURCE_GESTURE - will you be able to know gesture vs other dispatcher usages (and what are they? Profiles plugin only?)

Reply:
I just wanted some room here. Maybe i will delete that at the end.